packages/arb-token-bridge-ui/src/util/deposits/helpers.ts

```diff
@@ -265,6 +299,127 @@ const updateClassicDepositStatusData = async ({
   return completeDepositTx
 }
 
+export async function updateTeleporterDepositStatusData({
+  assetType,
+  parentChainId,
+  childChainId,
+  txID
+}: {
+  assetType: AssetType
+  parentChainId: number
+  childChainId: number
+  txID: string
+}): Promise<{
+  status?: TxnStatus
+  timestampResolved?: string
+  l1ToL2MsgData: L1ToL2MessageData
+  l2ToL3MsgData: L2ToL3MessageData
+}> {
+  const isNativeCurrencyTransfer = assetType === AssetType.ETH
+
+  const sourceChainProvider = getProvider(parentChainId)
+  const destinationChainProvider = getProvider(childChainId)
+
+  const { l2ChainId } = await getL2ConfigForTeleport({
+    destinationChainProvider
+  })
+
+  let depositStatus: Erc20DepositMessages | EthTeleportStatus | undefined,
+    l2Retryable,
+    l3Retryable,
+    completed,
+    l2TxHash,
+    l3TxHash,
+    l1ToL2MsgData: L1ToL2MessageData = {
+      status: L1ToL2MessageStatus.NOT_YET_CREATED,
+      fetchingUpdate: false
+    },
+    l2ToL3MsgData: L2ToL3MessageData = {
+      status: L1ToL2MessageStatus.NOT_YET_CREATED,
+      fetchingUpdate: false,
+      l2ChainId,
+      completed: false
+    }
```

Comment:
I don't like that we have 10 variables that we can reassign values to, feels very error prone. Better to early return than reassign

Reply:
doesn't seem right to initialise so many vars, there must be another way